lib/file-path-parser.js

```diff
@@ -0,0 +1,36 @@
+// Exports a factory for testing purposes.
+// Otherwise process will be the real deal.
+// Also, testing with time/timezones is rather difficult, so that's all mocked.
+module.exports = function(myProcess, myDate, myFormatter) {
+  return function(s) {
+    r = s;
```

Comment:
Can you please use better names here? It's hard for me to understand what happens. Also, i am not sure if this is needed, since `s` is not used anymore, or i didn't see it?